spec/index.bs

```diff
@@ -9856,36 +9856,29 @@ of WebGPU calls, typically for debugging purposes or to make an operation more f
     ::
         The first {{GPUError}}, if any, observed while the [=GPU error scope=] was current.
 
-    : <dfn>\[[filter]]</dfn> of type {{GPUErrorFilter}}.
+    : <dfn>\[[filter]]</dfn> of type {{GPUErrorType}}.
     ::
         Determines what type of {{GPUError}} this [=GPU error scope=] observes.
 </dl>
 
 <script type=idl>
-enum GPUErrorFilter {
+enum GPUErrorType {
     "out-of-memory",
     "validation",
 };
-</script>
 
-<script type=idl>
 [Exposed=(Window, DedicatedWorker), SecureContext]
-interface GPUOutOfMemoryError {
-    constructor();
-};
+interface GPUError {
+    constructor(GPUErrorType type, DOMString message);
 
-[Exposed=(Window, DedicatedWorker), SecureContext]
-interface GPUValidationError {
-    constructor(DOMString message);
+    readonly attribute GPUErrorType type;
     readonly attribute DOMString message;
```

Comment:
This change means that there could be a `message` on an out-of-memory error. It would be tempting for an implementation to put information here on what happened. But then apps will start parsing the info in that string to figure out how to recover.

So I'm considering writing something here that says `message` MUST be empty if the `type` is `"out-of-memory"`. If we want to start exposing more fine-grained error info about out-of-memory in the future, we can add a new field with predefined enum values that apps can reliably handle.

Reply:
The fine grained info I'm thinking is like, were we unable to allocate due to some hidden size limit or due to actually being physically unable to allocate. Thinking about it now that doesn't seem like that much of a concern with the resolution of #1371.

The note you suggest is a good idea though.